hnn_core/network.py

```diff
@@ -107,12 +108,15 @@ class Network(object):
 
     Attributes
     ----------
-    cellname_list : list
-        The names of real cell types in the network (e.g. 'L2_basket')
+    cell_types : dict
```

Comment:
there are [other places in the repo](https://github.com/jonescompneurolab/hnn-core/search?q=cell_types) where `cell_types` is a different kind of object. This could be potentially confusing

Reply:
`cell_templates` maybe?

Reply:
Given that I converted `cell_types` from `cell_list`, this attribute now serves two purposes: to document an exhaustive list of cells that are represented in the network (e.g., as in [here](https://github.com/jonescompneurolab/hnn-core/blob/32d33cc96bd788f6c96b97339a98788b2cf3810e/hnn_core/network.py#L720) or [here](https://github.com/jonescompneurolab/hnn-core/blob/32d33cc96bd788f6c96b97339a98788b2cf3810e/hnn_core/network.py#L697)), and also to specify the cell (class) template that will be used to represent each cell type. So if a new user or developer wants to see which cells are implemented in an instance of `Network`, (e.g., in conjunction with functions like `plot_cell_morphology()`, they should look towards `Network.cell_types` for guidance.

Toward your point though, maybe what we want to do is have a `cell_types` attribute of type list that mirrors `Network.cell_templates.keys()`?

Reply:
Right, and ignore `plot_cell_morphology` for now. I'm going to move that into the cell class, so that will become irrelevant. So, it's only in `cell_response.py` that we have a bit of mismatch. You could go either way -- rename that one instance of `cell_types` and/or add a `property` function that returns `cell_types` using `cell_templates.keys()`

Reply:
So you're thinking just rename `cell_types` [here](https://github.com/jonescompneurolab/hnn-core/blob/70a38d6b43a0801a88b4d8c4742c644704eed828/hnn_core/cell_response.py#L284)?

Reply:
yep!

Reply:
What do you think of https://github.com/jonescompneurolab/hnn-core/pull/321/commits/1fff52c13c14a4d4feaee379afa786a6e427c19e @jasmainak ? My thinking here was to also reduce the amount of hardcoding we have for cell type names. cc @ntolley

Reply:
I like it!